packages/components/src/components/teaser/teaser.ts

```diff
@@ -0,0 +1,170 @@
+import { css, html } from 'lit';
+import { customElement, property, query, state } from 'lit/decorators.js';
+import { HasSlotController } from '../../internal/slot';
+import cx from 'classix';
+import SolidElement from '../../internal/solid-element';
+import type { PropertyValues } from 'lit';
+
+/**
+ * @summary Teasers group information into flexible containers so users can browse a collection of related items and actions.
+ * @documentation https://solid.union-investment.com/[storybook-link]/teaser
+ *
+ * @status stable
+ * @since 1.2
+ * *
+ * @slot - An optional main content slot.
+ * @slot media - An optional media slot.
+ * @slot meta - An optional meta slot.
+ * @slot headline - headline slot.
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart media - The container that wraps the media.
+ * @csspart meta -  The container that wraps the meta.
+ * @csspart headline - The container that wraps the headline.
+ * @csspart main - The container that wraps the main content.
+ *
+ * @cssproperty --distribution-media - The distribution ratio of the media.
+ * @cssproperty --distribution-content - The distribution ratio of the content.
+ */
+
+@customElement('sd-teaser')
+export default class SdTeaser extends SolidElement {
+  @property({ reflect: true })
+  variant: 'white' | 'white border-neutral-300' | 'neutral-100' | 'primary' | 'primary-100' = 'white';
+
+  /** The teaser's orientation */
+  @property({ reflect: true })
+  orientation: 'responsive' | 'vertical' | 'horizontal' = 'responsive';
+
+  /** The teaser's inner padding. This is always set in `white border-neutral-300`. */
+  @property({ type: Boolean, reflect: true }) inset = false;
+
+  @query('[part="base"]') teaser: HTMLElement;
+
+  @state() private currentOrientation: 'vertical' | 'horizontal';
+
+  private readonly hasSlotController = new HasSlotController(this, '[default]', 'media', 'meta', 'headline');
+
+  private resizeObserver: ResizeObserver;
+
+  connectedCallback() {
+    super.connectedCallback();
+    if (window.ResizeObserver) {
+      this.resizeObserver = new ResizeObserver(() => this.updateOrientation());
+      this.resizeObserver.observe(this);
+    }
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this.resizeObserver) {
+      this.resizeObserver.disconnect();
+    }
+  }
+
+  updated(changedProperties: PropertyValues) {
+    super.updated(changedProperties);
+    if (changedProperties.has('orientation')) {
+      this.updateOrientation();
+    }
+  }
+
+  updateOrientation() {
+    if (this.orientation === 'responsive') {
+      this.currentOrientation = this.offsetWidth > 448 ? 'horizontal' : 'vertical';
```

Comment:
I'm just realizing: The component is missing the `breakpoint` property. The 448px should be overridable from outside.

Thinking about that... do you think we would need the `responsive` property as someone could just set the `breakpoint` to `0` for always vertical and to e. g. `9999` for always horizontal? Should we do a session on that together? :)